File: 1.21.9/src/main/java/com/bleudev/nine_lifes/custom/event/EntitySpawnEvents.java
package com.bleudev.nine_lifes.custom.event;


import net.fabricmc.fabric.api.event.Event;
import net.fabricmc.fabric.api.event.EventFactory;
import net.minecraft.server.level.ServerLevel;
import net.minecraft.world.entity.Entity;

public class EntitySpawnEvents {
    public static final Event<EntitySpawn> ENTITY_SPAWN = EventFactory.createArrayBacked(EntitySpawn.class, callbacks -> ((entity, level) -> {
        for (EntitySpawn callback : callbacks)
            callback.onEntitySpawn(entity, level);
    }));

    @FunctionalInterface
    public interface EntitySpawn {
        void onEntitySpawn(Entity entity, ServerLevel level);
    }
}

File: 1.21.9/src/main/kotlin/com/bleudev/nine_lifes/api/event/EntitySpawnEvents.kt
package com.bleudev.nine_lifes.api.event

import net.fabricmc.fabric.api.event.Event
import net.fabricmc.fabric.api.event.EventFactory
import net.minecraft.server.level.ServerLevel
import net.minecraft.world.entity.Entity

typealias EntitySpawn = (entity: Entity, level: ServerLevel) -> Unit

object EntitySpawnEvents {
    @JvmField
    val ENTITY_SPAWN: Event<EntitySpawn> = EventFactory.createArrayBacked(EntitySpawn::class.java) { callbacks ->
        { a, b -> for (callback in callbacks) callback(a, b) }
    }
}